2024-06-06 10:18:20,263 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Current SDK version is 0.15.8
2024-06-06 10:18:20,263 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Configure stats pid to 22424
2024-06-06 10:18:20,263 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Loading settings from C:\Users\ly\.config\wandb\settings
2024-06-06 10:18:20,263 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Loading settings from D:\Research-Projects-jupyter\PFLlib\system\wandb\settings
2024-06-06 10:18:20,263 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Loading settings from environment variables: {}
2024-06-06 10:18:20,263 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Inferring run settings from compute environment: {'program_relpath': 'system\\main.py', 'program': 'D:\\Research-Projects-jupyter\\PFLlib\\system\\main.py'}
2024-06-06 10:18:20,263 INFO    MainThread:22424 [wandb_init.py:_log_setup():507] Logging user logs to D:\Research-Projects-jupyter\PFLlib\system\wandb\run-20240606_101820-8uzld4po\logs\debug.log
2024-06-06 10:18:20,263 INFO    MainThread:22424 [wandb_init.py:_log_setup():508] Logging internal logs to D:\Research-Projects-jupyter\PFLlib\system\wandb\run-20240606_101820-8uzld4po\logs\debug-internal.log
2024-06-06 10:18:20,263 INFO    MainThread:22424 [wandb_init.py:_jupyter_setup():453] configuring jupyter hooks <wandb.sdk.wandb_init._WandbInit object at 0x000002658CF35B08>
2024-06-06 10:18:20,264 INFO    MainThread:22424 [wandb_init.py:init():547] calling init triggers
2024-06-06 10:18:20,264 INFO    MainThread:22424 [wandb_init.py:init():555] wandb.init called with sweep_config: {}
config: {'goal': 'test', 'device': 'cuda', 'device_id': '0', 'dataset': 'Cifar10', 'num_classes': 10, 'model': BaseHeadSplit(
  (base): FedAvgCNN(
    (conv1): Sequential(
      (0): Conv2d(3, 32, kernel_size=(5, 5), stride=(1, 1))
      (1): ReLU(inplace=True)
      (2): MaxPool2d(kernel_size=(2, 2), stride=(2, 2), padding=0, dilation=1, ceil_mode=False)
    )
    (conv2): Sequential(
      (0): Conv2d(32, 64, kernel_size=(5, 5), stride=(1, 1))
      (1): ReLU(inplace=True)
      (2): MaxPool2d(kernel_size=(2, 2), stride=(2, 2), padding=0, dilation=1, ceil_mode=False)
    )
    (fc1): Sequential(
      (0): Linear(in_features=1600, out_features=512, bias=True)
      (1): ReLU(inplace=True)
    )
    (fc): Identity()
  )
  (head): Linear(in_features=512, out_features=10, bias=True)
), 'batch_size': 64, 'local_learning_rate': 0.005, 'learning_rate_decay': False, 'learning_rate_decay_gamma': 0.99, 'global_rounds': 500, 'local_epochs': 1, 'algorithm': 'FedASCL', 'join_ratio': 1.0, 'random_join_ratio': False, 'num_clients': 20, 'prev': 0, 'times': 1, 'eval_gap': 10, 'save_folder_name': 'items', 'auto_break': False, 'dlg_eval': False, 'dlg_gap': 100, 'batch_num_per_client': 2, 'num_new_clients': 0, 'fine_tuning_epoch_new': 0, 'client_drop_rate': 0.0, 'train_slow_rate': 0.0, 'send_slow_rate': 0.0, 'time_select': False, 'time_threthold': 10000, 'beta': 0.0, 'lamda': 1.0, 'mu': 0.0, 'K': 5, 'p_learning_rate': 0.01, 'M': 5, 'itk': 4000, 'alphaK': 1.0, 'sigma': 1.0, 'alpha': 1.0, 'plocal_epochs': 1, 'tau': 1.0, 'fine_tuning_epochs': 10, 'dr_learning_rate': 0.0, 'L': 1.0, 'noise_dim': 512, 'generator_learning_rate': 0.005, 'hidden_dim': 512, 'server_epochs': 1000, 'localize_feature_extractor': False, 'server_learning_rate': 1.0, 'eta': 1.0, 'rand_percent': 80, 'layer_idx': 2, 'mentee_learning_rate': 0.005, 'T_start': 0.95, 'T_end': 0.98, 'momentum': 0.1, 'kl_weight': 0.0, 'wandb': 'SCL_test', 'margin_type': 'raw', 'model_str': 'cnn', 'head': Linear(in_features=512, out_features=10, bias=True)}
2024-06-06 10:18:20,264 INFO    MainThread:22424 [wandb_init.py:init():596] starting backend
2024-06-06 10:18:20,264 INFO    MainThread:22424 [wandb_init.py:init():600] setting up manager
2024-06-06 10:18:20,265 INFO    MainThread:22424 [backend.py:_multiprocessing_setup():108] multiprocessing start_methods=spawn, using: spawn
2024-06-06 10:18:20,267 INFO    MainThread:22424 [wandb_init.py:init():606] backend started and connected
2024-06-06 10:18:20,275 INFO    MainThread:22424 [wandb_run.py:_label_probe_notebook():1234] probe notebook
2024-06-06 10:18:20,276 INFO    MainThread:22424 [wandb_run.py:_label_probe_notebook():1244] Unable to probe notebook: 'NoneType' object has no attribute 'get'
2024-06-06 10:18:20,276 INFO    MainThread:22424 [wandb_init.py:init():697] updated telemetry
2024-06-06 10:18:20,316 INFO    MainThread:22424 [wandb_init.py:init():730] communicating run to backend with 60.0 second timeout
2024-06-06 10:18:21,081 INFO    MainThread:22424 [wandb_run.py:_on_init():2180] communicating current version
2024-06-06 10:18:21,526 INFO    MainThread:22424 [wandb_run.py:_on_init():2189] got version response upgrade_message: "wandb version 0.17.0 is available!  To upgrade, please run:\n $ pip install wandb --upgrade"

2024-06-06 10:18:21,526 INFO    MainThread:22424 [wandb_init.py:init():781] starting run threads in backend
2024-06-06 10:18:22,693 INFO    MainThread:22424 [wandb_run.py:_console_start():2159] atexit reg
2024-06-06 10:18:22,693 INFO    MainThread:22424 [wandb_run.py:_redirect():2014] redirect: wrap_raw
2024-06-06 10:18:22,694 INFO    MainThread:22424 [wandb_run.py:_redirect():2079] Wrapping output streams.
2024-06-06 10:18:22,694 INFO    MainThread:22424 [wandb_run.py:_redirect():2104] Redirects installed.
2024-06-06 10:18:22,695 INFO    MainThread:22424 [wandb_init.py:init():822] run started, returning control to user process
2024-06-06 10:58:07,392 INFO    MainThread:22424 [jupyter.py:_save_ipynb():384] looking for notebook: SCL_Exps-jvsc-f030b701-b833-46c7-8430-b07a1d98f284e88941e3-e4b2-461d-a227-e54f53d9515b.ipynb
2024-06-06 10:58:07,394 INFO    MainThread:22424 [wandb_init.py:_pause_backend():418] pausing backend
2024-06-06 10:58:07,420 INFO    MainThread:22424 [wandb_init.py:_resume_backend():423] resuming backend
2024-06-06 10:58:07,434 INFO    MainThread:22424 [jupyter.py:_save_ipynb():384] looking for notebook: SCL_Exps-jvsc-f030b701-b833-46c7-8430-b07a1d98f284e88941e3-e4b2-461d-a227-e54f53d9515b.ipynb
2024-06-06 10:58:07,434 INFO    MainThread:22424 [wandb_init.py:_pause_backend():418] pausing backend
2024-06-06 10:58:07,451 INFO    MainThread:22424 [wandb_init.py:_resume_backend():423] resuming backend
2024-06-06 10:58:07,489 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Current SDK version is 0.15.8
2024-06-06 10:58:07,489 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Configure stats pid to 22424
2024-06-06 10:58:07,489 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Loading settings from C:\Users\ly\.config\wandb\settings
2024-06-06 10:58:07,489 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Loading settings from D:\Research-Projects-jupyter\PFLlib\system\wandb\settings
2024-06-06 10:58:07,489 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Loading settings from environment variables: {}
2024-06-06 10:58:07,489 INFO    MainThread:22424 [wandb_setup.py:_flush():76] Inferring run settings from compute environment: {'program_relpath': 'system\\main.py', 'program': 'D:\\Research-Projects-jupyter\\PFLlib\\system\\main.py'}
2024-06-06 10:58:07,489 INFO    MainThread:22424 [wandb_init.py:_log_setup():507] Logging user logs to D:\Research-Projects-jupyter\PFLlib\system\wandb\run-20240606_105807-qgxixyfm\logs\debug.log
2024-06-06 10:58:07,489 INFO    MainThread:22424 [wandb_init.py:_log_setup():508] Logging internal logs to D:\Research-Projects-jupyter\PFLlib\system\wandb\run-20240606_105807-qgxixyfm\logs\debug-internal.log
2024-06-06 10:58:07,489 INFO    MainThread:22424 [wandb_init.py:init():547] calling init triggers
2024-06-06 10:58:07,490 INFO    MainThread:22424 [wandb_init.py:init():555] wandb.init called with sweep_config: {}
config: {'goal': 'test', 'device': 'cuda', 'device_id': '0', 'dataset': 'Cifar10', 'num_classes': 10, 'model': BaseHeadSplit(
  (base): FedAvgCNN(
    (conv1): Sequential(
      (0): Conv2d(3, 32, kernel_size=(5, 5), stride=(1, 1))
      (1): ReLU(inplace=True)
      (2): MaxPool2d(kernel_size=(2, 2), stride=(2, 2), padding=0, dilation=1, ceil_mode=False)
    )
    (conv2): Sequential(
      (0): Conv2d(32, 64, kernel_size=(5, 5), stride=(1, 1))
      (1): ReLU(inplace=True)
      (2): MaxPool2d(kernel_size=(2, 2), stride=(2, 2), padding=0, dilation=1, ceil_mode=False)
    )
    (fc1): Sequential(
      (0): Linear(in_features=1600, out_features=512, bias=True)
      (1): ReLU(inplace=True)
    )
    (fc): Identity()
  )
  (head): Linear(in_features=512, out_features=10, bias=True)
), 'batch_size': 64, 'local_learning_rate': 0.005, 'learning_rate_decay': False, 'learning_rate_decay_gamma': 0.99, 'global_rounds': 500, 'local_epochs': 1, 'algorithm': 'FedASCL', 'join_ratio': 1.0, 'random_join_ratio': False, 'num_clients': 20, 'prev': 0, 'times': 1, 'eval_gap': 10, 'save_folder_name': 'items', 'auto_break': False, 'dlg_eval': False, 'dlg_gap': 100, 'batch_num_per_client': 2, 'num_new_clients': 0, 'fine_tuning_epoch_new': 0, 'client_drop_rate': 0.0, 'train_slow_rate': 0.0, 'send_slow_rate': 0.0, 'time_select': False, 'time_threthold': 10000, 'beta': 0.0, 'lamda': 1.0, 'mu': 0.0, 'K': 5, 'p_learning_rate': 0.01, 'M': 5, 'itk': 4000, 'alphaK': 1.0, 'sigma': 1.0, 'alpha': 1.0, 'plocal_epochs': 1, 'tau': 1.0, 'fine_tuning_epochs': 10, 'dr_learning_rate': 0.0, 'L': 1.0, 'noise_dim': 512, 'generator_learning_rate': 0.005, 'hidden_dim': 512, 'server_epochs': 1000, 'localize_feature_extractor': False, 'server_learning_rate': 1.0, 'eta': 1.0, 'rand_percent': 80, 'layer_idx': 2, 'mentee_learning_rate': 0.005, 'T_start': 0.95, 'T_end': 0.98, 'momentum': 0.1, 'kl_weight': 0.0, 'wandb': 'SCL_test', 'margin_type': 'None', 'model_str': 'cnn', 'head': Linear(in_features=512, out_features=10, bias=True)}
2024-06-06 10:58:07,490 INFO    MainThread:22424 [wandb_init.py:init():572] re-initializing run, found existing run on stack: 8uzld4po
2024-06-06 10:58:07,492 INFO    MainThread:22424 [wandb_run.py:_finish():1894] finishing run paddeyzhang/SCL_test/8uzld4po
2024-06-06 10:58:07,529 ERROR   MainThread:22424 [jupyter.py:save_history():500] Unable to save ipython session history:
{'matches': ['NameError', 'Net', 'None', 'NotADirectoryError', 'NotImplemented', 'NotImplementedError'], 'cursor_end': 100, 'cursor_start': 99, 'metadata': {'_jupyter_types_experimental': [{'start': 99, 'end': 100, 'text': 'NameError', 'type': '<unknown>', 'signature': ''}, {'start': 99, 'end': 100, 'text': 'Net', 'type': '<unknown>', 'signature': ''}, {'start': 99, 'end': 100, 'text': 'None', 'type': '<unknown>', 'signature': ''}, {'start': 99, 'end': 100, 'text': 'NotADirectoryError', 'type': '<unknown>', 'signature': ''}, {'start': 99, 'end': 100, 'text': 'NotImplemented', 'type': '<unknown>', 'signature': ''}, {'start': 99, 'end': 100, 'text': 'NotImplementedError', 'type': '<unknown>', 'signature': ''}]}, 'status': 'ok'} is not valid under any of the given schemas

Failed validating 'oneOf' in notebook['properties']['data']['additionalProperties']:

On instance['data']['application/vnd.vscode.bg.execution.38.result']:
{'cursor_end': 100,
 'cursor_start': 99,
 'matches': ['NameError',
             'Net',
             'None',
             'NotADirectoryError',
             'NotImplemented',
             'NotImplementedError'],
 'metadata': {'_jupyter_types_experimental': [{'end': 100,
                                               'signature': '',
                                               'start': 99,
                                               'text': 'NameError',
                                               'type': '<unknown>'},
                                              {'end': 100,
                                               'signature': '',
                                               'start': 99,
                                               'text': 'Net',
                                               'type': '<unknown>'},
                                              {'end': 100,
                                               'signature': '',
                                               'start': 99,
                                               'text': 'None',
                                               'type': '<unknown>'},
                                              {'end': 100,
                                               'signature': '',
                                               'start': 99,
                                               'text': 'NotADirectoryError',
                                               'type': '<unknown>'},
                                              {'end': 100,
                                               'signature': '',
                                               'start': 99,
                                               'text': 'NotImplemented',
                                               'type': '<unknown>'},
                                              {'end': 100,
                                               'signature': '',
                                               'start': 99,
                                               'text': 'NotImplementedError',
                                               'type': '<unknown>'}]},
 'status': 'ok'}
2024-06-06 10:58:07,530 INFO    MainThread:22424 [jupyter.py:_save_ipynb():384] looking for notebook: SCL_Exps-jvsc-f030b701-b833-46c7-8430-b07a1d98f284e88941e3-e4b2-461d-a227-e54f53d9515b.ipynb
2024-06-06 10:58:07,530 INFO    MainThread:22424 [wandb_init.py:_jupyter_teardown():435] cleaning up jupyter logic
2024-06-06 10:58:07,530 INFO    MainThread:22424 [wandb_run.py:_atexit_cleanup():2128] got exitcode: 0
2024-06-06 10:58:07,530 INFO    MainThread:22424 [wandb_run.py:_restore():2111] restore
2024-06-06 10:58:07,530 INFO    MainThread:22424 [wandb_run.py:_restore():2117] restore done
2024-06-06 10:58:23,070 INFO    MainThread:22424 [wandb_run.py:_footer_history_summary_info():3481] rendering history
2024-06-06 10:58:23,071 INFO    MainThread:22424 [wandb_run.py:_footer_history_summary_info():3513] rendering summary
2024-06-06 10:58:23,077 INFO    MainThread:22424 [wandb_run.py:_footer_sync_info():3440] logging synced files
